### PR TITLE
[docs] app commands guide

### PR DESCRIPTION
**Pull request by ika2kki**
## Summary

app commands guide drafting
i have a built version hosted here to make it accessible if you want to read it https://saya.kessoku.moe/discord.py/guide/interactions/slash_commands.html
it can be slightly old if i forgot to update it after a commit.
so far, i am kinda happy with the content but not so much the overall writing tone

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/guide/interactions/slash_commands.rst [hunk 1]
```diff
@@ -0,0 +1,928 @@
...
+    @client.tree.command()
+    async def meow(interaction: discord.Interaction):
+        """Meow meow meow"""
+
+        await interaction.response.send_message("meow")
+
+Functions of this pattern are called callbacks, since their execution is
+relinquished to the library to be called later.
+
+Callbacks always have one parameter, ``interaction``,
```
Comment (Willy-C) [inline docs/guide/interactions/slash_commands.rst:77]: I think this wording "always have one parameter" might be confuse some people into thinking callbacks can have only a single parameter and nothing more, although additional parameters are covered later in this guide as well. Maybe "always have at least one parameter" instead?

File: docs/guide/interactions/slash_commands.rst [hunk 2]
```diff
@@ -0,0 +1,941 @@
...
+-----------
+
+To work with app commands, bots need the ``applications.commands`` scope.
+
+You can enable this scope when generating an OAuth2 URL for your bot, shown :ref:`here <discord_invite_bot>`.
+
+Defining a Tree
+++++++++++++++++
+
+First, a :class:`.app_commands.CommandTree` needs to be instaniated
```
Comment (Jerrie-Aries) [inline docs/guide/interactions/slash_commands.rst:30]:
```suggestion
First, a :class:`.app_commands.CommandTree` needs to be instantiated
```
Fix typo.

**Review by Jerrie-Aries — CHANGES_REQUESTED** [review]

File: docs/guide/interactions/slash_commands.rst [hunk 3]
```diff
@@ -0,0 +1,1232 @@
...
+            command_prefix="?",
+            intents=discord.Intents.default(),
+            tree_cls=MyCommandTree
+        )
+
+Error handling
+---------------
+
+So far, any exceptions raised within a command callback, any custom checks or in a transformer should just be
+printed out in the program's ``stderr`` or through any custom logging handlers.
```
Comment (Willy-C) [inline docs/guide/interactions/slash_commands.rst:891]:
```suggestion
By default, exceptions are logged using the library logger if the command does not have any error handlers attached to it.
```
Errors are no longer printed to `stderr` by default since v2.0

File: docs/guide/interactions/slash_commands.rst [hunk 4]
```diff
@@ -0,0 +1,1232 @@
...
+        :param a: left operand
+        :param b: right operand
+        """
+
+If both are used, :func:`.app_commands.describe` always takes precedence.
+
+Naming
+^^^^^^^
+
+Since parameter names are confined to the rules of Python's syntax,
```
Comment (tailoric) [inline docs/guide/interactions/slash_commands.rst:387]: Slash command parameter also have confined rules, I feel like they should be mentioned in here too.
https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-naming

File: docs/guide/interactions/slash_commands.rst [hunk 5]
```diff
@@ -0,0 +1,1232 @@
...
+Choices
+++++++++
+
+To provide the user with a list of options to choose from for an argument, the :func:`.app_commands.choices` decorator can be applied.
+
+A user is restricted to selecting a choice and can't type something else.
+
+Each individual choice contains 2 fields:
+
+- A name, which is what the user sees
```
Comment (tailoric) [inline docs/guide/interactions/slash_commands.rst:425]:
```suggestion
- A name, which will be displayed to the user in the client.
```

File: docs/guide/interactions/slash_commands.rst [hunk 6]
```diff
@@ -0,0 +1,1232 @@
...
+- A value, which is hidden to the user and only visible to the API. Typically, this is either the same as the name or something more developer-friendly. Value types are limited to either a :class:`str`, :class:`int` or :class:`float`.
```
Comment (tailoric) [inline docs/guide/interactions/slash_commands.rst:426]: Not 100% sure about my own suggested wording here, but I think it is important to make clear that it is the actual value that gets given to the command by discord.
```suggestion
- A value, which will be the real value provided to the command callback by Discord. Typically, this is either the same as the name or something more developer-friendly. Value types are limited to either a :class:`str`, :class:`int` or :class:`float`.
```

File: docs/guide/interactions/slash_commands.rst [hunk 7]
```diff
@@ -0,0 +1,1232 @@
...
+- An autocomplete callback is called with the current user input.
+
+- Returned choices are sent back to Discord and shown in the user's client.
+
+  - An empty list can be returned to denote no choices.
+
+Attaching an autocomplete function to a parameter can be done in 2 main ways:
+
+1. From the command, with the :meth:`~.app_commands.Command.autocomplete` decorator
+2. With a separate decorator, :func:`.app_commands.autocomplete`
```
Comment (tailoric) [inline docs/guide/interactions/slash_commands.rst:474]: It is also possible to attach an autocompletion via a [Transformer](https://discordpy.readthedocs.io/en/latest/interactions/api.html?highlight=transformer%20autocomplete#discord.app_commands.Transformer.autocomplete)

File: docs/guide/interactions/slash_commands.rst [hunk 8]
```diff
@@ -0,0 +1,1232 @@
...
+User parameter
+^^^^^^^^^^^^^^^
+
+Annotating to either :class:`discord.User` or :class:`discord.Member` both point to a ``USER`` Discord-type.
+
+The actual type given by Discord is dependent on whether the command was invoked in DM-messages or in a guild.
+
+For example, if a parameter annotates to :class:`~discord.Member`, and the command is invoked in a guild,
+discord.py will raise an error since the actual type given by Discord,
+:class:`~discord.User`, is incompatible with :class:`~discord.Member`.
```
Comment (tailoric) [inline docs/guide/interactions/slash_commands.rst:609]:
```suggestion
For example, if a parameter annotates to :class:`~discord.Member`, and the command is invoked in a DM,
discord.py will raise an error since the actual type given by Discord,
:class:`~discord.User`, is incompatible with :class:`~discord.Member`.
```

File: docs/guide/interactions/slash_commands.rst [hunk 9]
```diff
@@ -0,0 +1,1232 @@
...
+.. code-block:: python
+
+    class Todo(app_commands.Group, name="todo", description="manages a todolist"):
+        ...
+
+    client.tree.add_command(Todo()) # required!
+
+.. note::
+
+    Groups need to be added to the command tree manually with :meth:`.CommandTree.add_command`,
```
Comment (tailoric) [inline docs/guide/interactions/slash_commands.rst:669]: What about [GroupCogs](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html?highlight=groupcog#discord.ext.commands.GroupCog)

Also within a cog they are also automatically added 
https://gist.github.com/AbstractUmbra/a9c188797ae194e592efe05fa129c57f#file-07-extension_with_optional_group-py

Or are Cogs out of scope for this guide?

File: docs/guide/interactions/slash_commands.rst [hunk 10]
```diff
@@ -0,0 +1,1232 @@
...
+            await interaction.response.send_message(f"hello i am {newname} now")
+
+Commands with this check are still visible and invocable in the bot's direct messages,
+regardless of the permissions specified.
+
+To prevent this, :func:`~.app_commands.guild_only` can also be added.
+
+.. warning::
+
+    This can be overriden to a different set of permissions by server administrators through the "Integrations" tab on the official client,
```
Comment (tailoric) [inline docs/guide/interactions/slash_commands.rst:836]:
```suggestion
    This can be overridden to a different set of permissions by server administrators through the "Integrations" tab on the official client,
```

**Review by tailoric — COMMENTED** [review]
Read through it once and annotated what I've found and gave some opinions 
However, I skimmed the translation part only, since I have no experience in that area.

File: docs/guide/interactions/slash_commands.rst (hunk 6 above)
Comment (tailoric) [reply docs/guide/interactions/slash_commands.rst]: Yeah, shows how little I actually work with choices vs. auto-completion. 
I incorrectly assumed it was analogous to auto-completion, which gives you only the actual value of the choice. 

I still find the phrasing of "only visible to the API" sub-optimal, since it reads it to me like it is only visible to the discord api, not your bot, at least in my reading of me putting myself in the position of someone new to the concepts of discord.py or slash commands.
Maybe rephrase to "only visible to your bot" or "only visible to your command callback". Even though I am also iffy on that phrasing entirely. 
just throwing out one last idea here:
```
A value, that should represent the actual value we want to use inside our command callback. 
This can either be the same as the name or something "machine readable"
- like an integer value (see example below)
- a column id in the database
- a mapping key for an internal dictionary
- or any other representation we need.
```

File: docs/guide/interactions/slash_commands.rst (hunk 9 above)
Comment (tailoric) [reply docs/guide/interactions/slash_commands.rst:685]: Okay, then I agree with the original phrasing and this feedback can be set to resolved 👍

File: docs/guide/interactions/slash_commands.rst [hunk 11]
```diff
@@ -0,0 +1,1254 @@
...
+regardless of the permissions specified.
+
+To prevent this, :func:`~.app_commands.guild_only` can also be added.
+
+.. warning::
+
+    This can be overridden to a different set of permissions by server administrators through the "Integrations" tab on the official client,
+    meaning, an invoking user might not actually have the permissions specified in the decorator.
+
+Custom checks
```
Comment (No767) [inline docs/guide/interactions/slash_commands.rst:859]: the context menus, checks and cooldown guide is written and waiting on the final review (https://github.com/Rapptz/discord.py/pull/7952/). You could probably just link the checks guide page to this section or wait and leave this section as-is.

Comment (No767) [reply docs/guide/interactions/slash_commands.rst:887]: sounds good!

File: docs/guide/interactions/slash_commands.rst [hunk 12]
```diff
@@ -0,0 +1,1519 @@
...
+        forecast = random.choice(weathers)
+
+        await interaction.followup.send(f'the weather today is {forecast}!')
+
+Syncing
+++++++++
+
+In order for this command to show up on Discord, the API needs some information regarding it, namely:
+
+- The name and description
```
Comment (No767) [inline docs/guide/interactions/slash_commands.rst:179]: Also, sync when you add or modify permissions (`default_permissions` and `nsfw`). It might be useful to denote when not to sync as well (pretty much if any of the code after the `async def ():` part, and library side checks)

File: docs/guide/interactions/slash_commands.rst [hunk 13]
```diff
@@ -0,0 +1,1519 @@
...
++++++++++++
+
+Descriptions are added to parameters using the :func:`.app_commands.describe` decorator,
+where each keyword is treated as a parameter name.
+
+.. code-block:: python
+
+    @client.tree.command()
+    @app_commands.describe(
+        liquid="what type of liquid is on the wall",
```
Comment (No767) [inline docs/guide/interactions/slash_commands.rst:351]: Noticed that the param descriptions were not capitalized, since the ones in the examples are. Also missing the command description from what I can tell.

```suggestion
        liquid="What type of liquid is on the wall",
        amount="How much of it is on the wall"
```

File: docs/guide/interactions/slash_commands.rst [hunk 14]
```diff
@@ -0,0 +1,1519 @@
...
+
+.. code-block:: python
+
+    @client.tree.command() # numpy
+    async def addition(interaction: discord.Interaction, a: int, b: int):
+        """adds 2 numbers together.
+
+        Parameters
+        -----------
+        a: int
```
Comment (No767) [inline docs/guide/interactions/slash_commands.rst:376]: Pretty much the same exact thing as the last comment (command and param descriptions not being capitalized). This pretty much applies to the other two docstrings below.

File: docs/guide/interactions/slash_commands.rst [hunk 15]
```diff
@@ -0,0 +1,1519 @@
...
+
+    Instead, an empty list is returned to the user.
+
+Range
+++++++
+
+:class:`str`, :class:`int` and :class:`float` type parameters can optionally set a minimum and maximum value.
+For strings, this limits the character count, whereas for numeric types this limits the magnitude.
+
+To set a range, a parameter should annotate to :class:`.app_commands.Range`.
```
Comment (No767) [inline docs/guide/interactions/slash_commands.rst:522]: It would be useful to have an example of how you would annotate it

**Review by No767 — CHANGES_REQUESTED** [review]

File: docs/guide/interactions/slash_commands.rst (hunk 15 above)
Comment (No767) [reply docs/guide/interactions/slash_commands.rst:522]: Makes sense 👍

File: docs/guide/interactions/slash_commands.rst (hunk 13 above)
Comment (No767) [reply docs/guide/interactions/slash_commands.rst:351]: Yea this makes total sense

File: docs/guide/interactions/slash_commands.rst [hunk 16]
```diff
@@ -0,0 +1,1519 @@
...
+.. image:: /images/guide/app_commands/bottles_command_described.png
+
+Not specifying a description results with an ellipsis "..." being used instead.
+
+In addition to the decorator, parameter descriptions can also be added using
+Google, Sphinx or Numpy style docstrings.
+
+Examples using a command to add 2 numbers together:
+
+.. code-block:: python
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:368]: this code block could use the inline tabs ext

File: docs/guide/interactions/slash_commands.rst [hunk 17]
```diff
@@ -0,0 +1,1548 @@
...
+
+        await interaction.response.send_message('meow')
+
+Functions of this pattern are called callbacks, since their execution is
+left to the library to be called later.
+
+There are two main decorators to use when creating a command:
+
+1. :meth:`tree.command() <.CommandTree.command>` (as seen above)
+2. :func:`.app_commands.command`
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:86]:
```suggestion
1. :meth:`@tree.command() <.CommandTree.command>` (as seen above)
2. :func:`@app_commands.command() <.app_commands.command>`
```
nice to be consistent (also think the @ is a nice touch)

File: docs/guide/interactions/slash_commands.rst [hunk 18]
```diff
@@ -0,0 +1,1548 @@
...
+    async def avatar(interaction: discord.Interaction, user: Optional[discord.User] = None):
+        avatar = (user or interaction.user).display_avatar
+        await interaction.response.send_message(avatar.url)
+
+On Discord:
+
+.. image:: /images/guide/app_commands/avatar_command_optional_preview.png
+    :width: 300
+
+`Python version 3.10+ union types <https://peps.python.org/pep-0604/>`_ are also supported instead of :obj:`typing.Optional`.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:291]:
```suggestion
:pep:`Python version 3.10+ union types <604>`_ are also supported instead of :obj:`typing.Optional`.
```
Again I wouldn't link to a PEP but this is slightly cleaner syntax, https://docs.python.org/3/library/stdtypes.html#types-union might be a better link.

File: docs/guide/interactions/slash_commands.rst [hunk 19]
```diff
@@ -0,0 +1,1548 @@
...
+    Union types can't mix Discord types.
+
+    Something like ``Union[discord.Member, discord.TextChannel]`` isn't possible.
+
+Refer to the :ref:`type conversion table <type_conversion>` for full information on sub-types.
+
+Describing
++++++++++++
+
+Descriptions are added to parameters using the :func:`.app_commands.describe` decorator,
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:351]:
```suggestion
Descriptions are added to parameters using the :func:`@app_commands.describe() <.app_commands.describe>` decorator,
```

File: docs/guide/interactions/slash_commands.rst [hunk 20]
```diff
@@ -0,0 +1,1548 @@
...
+
+            Parameters
+            -----------
+            a: int
+                left operand
+            b: int
+                right operand
+            """
+
+            await interaction.response.send_message(f'{a} + {b} is {a + b}!')
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:391]:
```suggestion
            await interaction.response.send_message(f'{a + b = }!')
```
Not too sure about whether or not you'd want to show this, up to you

File: docs/guide/interactions/slash_commands.rst [hunk 21]
```diff
@@ -0,0 +1,1548 @@
...
+            :param a: left operand
+            :param b: right operand
+            """
+
+            await interaction.response.send_message(f'{a} + {b} is {a + b}!')
+
+Other meta info can be specified in the docstring, such as the function return type,
+but in-practice only the parameter descriptions are used.
+
+Parameter descriptions added using :func:`.app_commands.describe` always
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:425]:
```suggestion
Parameter descriptions added using :func:`@app_commands.describe() <.app_commands.describe>` always
```

File: docs/guide/interactions/slash_commands.rst [hunk 22]
```diff
@@ -0,0 +1,1548 @@
...
+take precedence over ones specified in the docstring.
+
+Naming
+^^^^^^^
+
+Since parameter names are confined to the rules of Python's syntax,
+the library offers a method to rename them with the :func:`.app_commands.rename` decorator.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:432]:
```suggestion
the library offers a method to rename them with the :func:`@app_commands.rename() <.app_commands.rename>` decorator.
```

File: docs/guide/interactions/slash_commands.rst [hunk 23]
```diff
@@ -0,0 +1,1548 @@
...
+        amount='how much of it is on the wall'
+    )
+    async def bottles(interaction: discord.Interaction, liquid: str, amount: int):
+        await interaction.response.send_message(f'{amount} bottles of {liquid} on the wall!')
+
+Choices
+++++++++
+
+:class:`str`, :class:`int` and :class:`float` type parameters can optionally set a list of choices for an argument
+using the :func:`.app_commands.choices` decorator.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:461]:
```suggestion
using the :func:`app_commands.choices() <.app_commands.choices>` decorator.
```

File: docs/guide/interactions/slash_commands.rst [hunk 24]
```diff
@@ -0,0 +1,1548 @@
...
+- An autocomplete callback is called with the current user input.
+
+- Returned choices are sent back to Discord and shown in the user's client.
+
+  - An empty list can be returned to denote no choices.
+
+Attaching an autocomplete function to a parameter can be done in 2 main ways:
+
+1. From the command, with the :meth:`~.app_commands.Command.autocomplete` decorator
+2. With a separate decorator, :func:`.app_commands.autocomplete`
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:524]:
```suggestion
1. From the command, with the :meth:`@Command.autocomplete() <.app_commands.Command.autocomplete>` decorator
2. With a separate decorator, :func:`@app_commands.autocomplete() <.app_commands.autocomplete>`
```
(not sure if the class name should show up here)

File: docs/guide/interactions/slash_commands.rst [hunk 25]
```diff
@@ -0,0 +1,1548 @@
...
+
+Code examples for either method can be found in the corresponding reference page.
+
+.. note::
+
+    Unlike :func:`.app_commands.choices`, a user can still submit any value instead of
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:530]:
```suggestion
    Unlike :func:`@app_commands.choices <.app_commands.choices>`, a user can still submit any value instead of
```

File: docs/guide/interactions/slash_commands.rst [hunk 26]
```diff
@@ -0,0 +1,1548 @@
...
+
+These can be provided by overriding the following properties:
+
+- :attr:`~.Transformer.type`
+- :attr:`~.Transformer.min_value`
+- :attr:`~.Transformer.max_value`
+- :attr:`~.Transformer.choices`
+- :attr:`~.Transformer.channel_types`
+
+Since these are properties, they must be decorated with :class:`property`:
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:611]:
```suggestion
Since these are properties, they must be decorated with :class:`@property <property>`:
```

File: docs/guide/interactions/slash_commands.rst [hunk 27]
```diff
@@ -0,0 +1,1548 @@
...
+    │   └── /todo lists switch
+    ├── /todo add
+    └── /todo delete
+
+Command groups **are not invocable** on their own.
+
+Therefore, instead of creating a command the standard way by decorating an async function,
+groups are created by using :class:`.app_commands.Group`.
+
+This class is customisable by subclassing and passing in any relevant fields at inheritance:
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:717]:
```suggestion
This class can be subclassed and attributes passed to the class constructor:
```
IIRC this is the "class constructor" but I can't find anything linking to it 😎

File: docs/guide/interactions/slash_commands.rst [hunk 28]
```diff
@@ -0,0 +1,1548 @@
...
+
+    class Todo(app_commands.Group, description='manages a todolist'):
+        ...
+
+    client.tree.add_command(Todo()) # required!
+
+.. note::
+
+    Groups need to be added to the command tree manually with :meth:`.CommandTree.add_command`,
+    since we lose the shortcut decorator :meth:`.CommandTree.command` with this class approach.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:729]:
```suggestion
    since we lose the shortcut decorator :meth:`@CommandTree.command() <.CommandTree.command>` with this class approach.
```

File: docs/guide/interactions/slash_commands.rst [hunk 29]
```diff
@@ -0,0 +1,1548 @@
...
+            ...
+
+    class Todo(app_commands.Group, description='manages a todolist'):
+        @app_commands.command(name='add', description='add a todo')
+        async def todo_add(self, interaction: discord.Interaction):
+            await interaction.response.send_message('added something to your todolist...!')
+
+        todo_lists = TodoLists()
+
+Decorators like :func:`.app_commands.default_permissions` and :func:`.app_commands.guild_only`
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:789]:
```suggestion
Decorators like :func:`@app_commands.default_permissions() <.app_commands.default_permissions>` and :func:`@app_commands.guild_only() <.app_commands.guild_only>`
```

File: docs/guide/interactions/slash_commands.rst [hunk 30]
```diff
@@ -0,0 +1,1548 @@
...
+---------------
+
+So far, all the command examples in this page have been global commands,
+which every guild your bot is in can see and use.
+
+In contrast, guild commands are only seeable and usable by members of a certain guild.
+
+There are 2 main ways to specify which guilds a command should sync a copy to:
+
+- Via the :func:`.app_commands.guilds` decorator, which takes a variadic amount of guilds
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:812]:
```suggestion
- Via the :func:`@app_commands.guilds() <.app_commands.guilds>` decorator, which takes a variadic amount of guilds
```

File: docs/guide/interactions/slash_commands.rst [hunk 31]
```diff
@@ -0,0 +1,1548 @@
...
+    @client.tree.command(nsfw=True)
+    async def evil(interaction: discord.Interaction):
+        await interaction.response.send_message('******') # very explicit text!
+
+Guild-only
++++++++++++
+
+Indicates whether this command can only be used in guilds or not.
+
+Enabled by adding the :func:`.app_commands.guild_only` decorator when defining an app command:
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:887]:
```suggestion
Enabled by adding the :func:`@app_commands.guilds() <.app_commands.guild_only>` decorator when defining an app command:
```

File: docs/guide/interactions/slash_commands.rst [hunk 32]
```diff
@@ -0,0 +1,1548 @@
...
+    async def serverinfo(interaction: discord.Interaction):
+        assert interaction.guild is not None
+        await interaction.response.send_message(interaction.guild.name)
+
+Default permissions
+++++++++++++++++++++
+
+This sets the default permissions a user needs in order to be able to see and invoke an app command.
+
+Configured by adding the :func:`.app_commands.default_permissions` decorator when defining an app command:
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:902]:
```suggestion
Configured by adding the :func:`@app_commands.default_permissions() <.app_commands.default_permissions>` decorator when defining an app command:
```

File: docs/guide/interactions/slash_commands.rst [hunk 33]
```diff
@@ -0,0 +1,1548 @@
...
+        if not guild:
+            await interaction.response.send_message("i can't change my name here")
+        else:
+            await guild.me.edit(nick=newname)
+            await interaction.response.send_message(f'hello i am {newname} now')
+
+Commands with this check are still visible and invocable in the bot's direct messages,
+regardless of the permissions specified.
+
+To prevent this, :func:`~.app_commands.guild_only` can also be added.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:919]:
```suggestion
To prevent this, :func:`@guild_only() <.app_commands.guild_only>` can also be added.
```

File: docs/guide/interactions/slash_commands.rst [hunk 34]
```diff
@@ -0,0 +1,1548 @@
...
+    async def fiftyfifty(interaction: discord.Interaction):
+        await interaction.response.send_message("you're very patient and lucky!")
+
+Custom checks can either be:
+
+- local, only running for a single command (as seen above).
+
+- on a group, running for all child commands, and before any local checks.
+
+ - Added using the :meth:`.app_commands.Group.error` decorator or overriding :meth:`.app_commands.Group.on_error`.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:999]:
```suggestion
 - Added using the :meth:`@app_commands.Group.error() <.app_commands.Group.error>` decorator or overriding :meth:`.app_commands.Group.on_error`.
```

File: docs/guide/interactions/slash_commands.rst [hunk 35]
```diff
@@ -0,0 +1,1548 @@
...
+            tree_cls=CoolPeopleTree
+        )
+
+.. _error_handling:
+
+Error handling
+---------------
+
+So far, any exceptions raised within a command callback, any custom checks, in a transformer
+or during localisation, et cetera should just be logged in the program's ``stderr`` or through any custom logging handlers.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:1055]:
```suggestion
or during localisation, et cetera should just be logged in the program's :obj:`sys.stderr` or through any custom logging handlers.
```

File: docs/guide/interactions/slash_commands.rst [hunk 36]
```diff
@@ -0,0 +1,1548 @@
...
+
+In order to catch exceptions and do something else, such as sending a message to let
+a user know their invocation failed for some reason, the library uses something called error handlers.
+
+There are 3 types of handlers:
+
+1. A local handler, which only catches exceptions for a specific command
+
+   Attached using the :meth:`.app_commands.Command.error` decorator.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:1064]:
```suggestion
   Attached using the :meth:`@app_commands.Command.error() <.app_commands.Command.error>` decorator.
```

File: docs/guide/interactions/slash_commands.rst [hunk 37]
```diff
@@ -0,0 +1,1548 @@
...
+
+2. A group handler, which catches exceptions only for a certain group's subcommands.
+
+   Added by using the :meth:`.app_commands.Group.error` decorator or overriding :meth:`.app_commands.Group.on_error`.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:1068]:
```suggestion
   Added by using the :meth:`@app_commands.Group.error() <.app_commands.Group.error>` decorator or overriding :meth:`.app_commands.Group.on_error`.
```

File: docs/guide/interactions/slash_commands.rst [hunk 38]
```diff
@@ -0,0 +1,1548 @@
...
+
+3. A global handler, which catches all exceptions in all commands.
+
+   Added by using the :meth:`.CommandTree.error` decorator or overriding :meth:`.CommandTree.on_error`.
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:1072]:
```suggestion
   Added by using the :meth:`@CommandTree.error() <.CommandTree.error>` decorator or overriding :meth:`.CommandTree.on_error`.
```

File: docs/guide/interactions/slash_commands.rst [hunk 39]
```diff
@@ -0,0 +1,1548 @@
...
+            return when
+
+    # pretend `some_command` is a command that uses this transformer
+
+    @some_command.error
+    async def some_command_error(interaction: discord.Interaction, error: app_commands.AppCommandError):
+        if isinstance(error, BadDateArgument):
+            await interaction.response.send_message(str(error))
+
+Instead of printing plainly to :obj:`sys.stderr`, the standard ``logging`` module can be configured instead -
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:1201]:
```suggestion
Instead of printing plainly to :obj:`sys.stderr`, the standard :mod:`logging` module can be configured instead -
```

File: docs/guide/interactions/slash_commands.rst [hunk 40]
```diff
@@ -0,0 +1,1548 @@
...
+
+1. ``string`` - the string to be translated according to ``locale``
+2. ``locale`` - the locale to translate to
+3. ``context`` - the context of this translation (what type of string is being translated)
+
+When :meth:`.CommandTree.sync` is called, this method is called in a heavy loop for each
+string for each locale.
+
+A wide variety of translation systems can be implemented using this interface, such as
+`gettext <https://docs.python.org/3/library/gettext.html>`_ and
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:1237]:
```suggestion
:mod:`gettext` and
```

File: docs/guide/interactions/slash_commands.rst [hunk 41]
```diff
@@ -0,0 +1,1548 @@
...
+        ...
+
+To toggle this behaviour, set the ``auto_locale_strings`` keyword-argument
+to :obj:`False` when creating a command:
+
+.. code-block:: python
+
+    @client.tree.command(name='example', description='an example command', auto_locale_strings=False)
+    async def example(interaction: discord.Interaction):
+        # i am not translated
```
Comment (Gobot1234) [inline docs/guide/interactions/slash_commands.rst:1261]:
```suggestion
        ...  # i am not translated
```

**Review by Gobot1234 — COMMENTED** [review]
Feel free to ignore the comments about decorators (IDR what other guides do)

File: docs/guide/interactions/slash_commands.rst [hunk 42]
```diff
@@ -0,0 +1,1618 @@
...
++=================+====================================================================================+
+| ``STRING``      | :class:`str`                                                                       |
++-----------------+------------------------------------------------------------------------------------+
+| ``INTEGER``     | :class:`int`                                                                       |
++-----------------+------------------------------------------------------------------------------------+
+| ``BOOLEAN``     | :class:`bool`                                                                      |
++-----------------+------------------------------------------------------------------------------------+
+| ``NUMBER``      | :class:`float`                                                                     |
++-----------------+------------------------------------------------------------------------------------+
+| ``USER``        | :class:`~discord.User` or :class:`~discord.Member`                                 |
```
Comment (No767) [inline docs/guide/interactions/slash_commands.rst:743]: I feel like it's worth noting that when using `discord.Member` as a typehint, that it will not have any presence data (e.g. there is no data on the online status of an member)

**Review by No767 — CHANGES_REQUESTED** [review]
Everything looks pretty good to me. There are a couple of minor phrasing issues that should be pointed out

File: docs/guide/interactions/slash_commands.rst (hunk 42 above)
Comment (No767) [reply docs/guide/interactions/slash_commands.rst:772]: Thinking about it, you are right about it

**Review by No767 — APPROVED** [review]
Just looked at the recent changes, and they all look good to me.

**Comment by ADHIRAJ191207** [comment]
maybe time to renovate it!
